src/OrbitGgp/ProjectTest.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <qvector.h>
+
+#include <QString>
+
+#include "OrbitGgp/Project.h"
+#include "TestUtils/TestUtils.h"
+
+namespace orbit_ggp {
+
+TEST(OrbitGgpProject, GetListFromJson) {
+  {
+    // invalid json
+    const auto json = QString("json").toUtf8();
+    EXPECT_THAT(Project::GetListFromJson(json), orbit_test_utils::HasError("Unable to parse JSON"));
+  }
+
+  {
+    // empty json
+    const auto json = QString("[]").toUtf8();
+    const auto empty_instances = Project::GetListFromJson(json);
+    ASSERT_THAT(empty_instances, orbit_test_utils::HasValue());
+    EXPECT_TRUE(empty_instances.value().empty());
+  }
+
+  {
+    // one empty json object
+    const auto json = QString("[{}]").toUtf8();
+    EXPECT_THAT(Project::GetListFromJson(json), orbit_test_utils::HasError("Unable to parse JSON"));
+  }
+
+  {
+    // one partial (invalid) element
+    // pretty json:
+    // [
+    //  {
+    //   "displayName": "display name"
+    //  }
+    // ]
+    const auto json = QString(R"([{"displayName":"display name"}])").toUtf8();
```

Comment:
Isn't cleaner to have a variable name with that json and try to parse it directly? 

(Something like that but I'm not sure how to write a multi-line string)
```
const Qstring kOnePartialInvalidElement = (R"(
     [
      {
       "displayName": "display name"
      }
    ])"
const auto json = kOnePartialInvalidElement.toUtf8();
```

Reply:
I changed it to not have the pretty version in the comment, but the json string being pretty json on its own. It messes a bit with the formatting but its better than before. Thanks for the suggestion!

Reply:
So sorry for the formatting. :(